cozy/main.py

```diff
@@ -167,6 +168,11 @@ def callback(res):
     share_info = defaultdict(list)
 
     try:
+        ## TODO: Is it OK for both `java` and `cxx` to be `None`?  This
+        ## code permits it, but I'm not sure that makes sense from a user
+        ## perspective.  Relatedly, should the variable be an enum rather
+        ## than two booleans?  (Though that would require another variable
+        ## for the file name...)
```

Comment:
This is a reasonable suggestion, but I'm not inclined to act on it. The current design is very simple, and therefore I think very understandable: `--java`, `--c++`, and `--save` are actions that Cozy can take after synthesis. It can take any subset of them (even none of them).

We use the `--save` flag without either `--java` or `--c++` insulate our own evaluation infrastructure from codegen bugs by preserving Cozy's final intermediate result. Synthesis and codegen are done in two completely separate invocations of Cozy. It is simply too frustrating to run Cozy for 3 hours only to have it fail in the last few seconds because of a codegen bug.

Reply:
OK.  I didn't realize about the `--save` option, so that is reasonable.